src/snakemake_interface_common/plugin_registry/plugin.py

```
@@ -250,7 +253,7 @@ def extract_values(value, thefield, name, tag=None):
                         value = self._parse_type(
                             thefield, value, thefield.metadata["type"]
                         )
-                    elif thefield.type != str:
+                    elif thefield.type != str and isinstance(thefield.type, type):
```

Ruff (0.8.2)

Use `is` and `is not` for type comparisons, or `isinstance()` for isinstance checks

(E721)